eclipse-theia/theia
# Support VSCode WebviewView API

#### What it does

Closes https://github.com/eclipse-theia/theia/issues/8740
Closes https://github.com/eclipse-theia/theia/pull/10137

Implements the VSCode WebviewView API mostly based on the changes done by #10137.

- [x] CQ for [`webview-views-main.ts`](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=23825)

#### How to test

1. Install the [`webview-view-sample`](https://github.com/microsoft/vscode-extension-samples/tree/main/webview-view-sample) extension. (download [here](https://github.com/eclipse-theia/theia/files/7988388/calico-colors-0.0.1.zip))
2. Open the `Calico Colors` container in the explorer view.
3. Assert that everything in there works as expected (opening/collapsing the view, adding colors, clearing colors)

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
